drivers/sht2x/sht2x.c

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (C) 2017 George Psimenos (gp7g14@soton.ac.uk)
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_sht2x
+ * @{
+ *
+ * @file
+ * @brief       Driver for the Sensirion SHT2x sensor series
+ *
+ * @author      George Psimenos <gp7g14@soton.ac.uk>
+ *
+ * @}
+ */
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <string.h>
+
+#include "assert.h"
+#include "periph/i2c.h"
+#include "xtimer.h"
+#include "sht2x.h"
+#include "sht2x_regs.h"
+
+#define ENABLE_DEBUG        (0)
+#include "debug.h"
+
+#define I2C_SPEED           I2C_SPEED_FAST
+#define MAX_RETRIES         20
+
+#define BUS                 (dev->params.i2c)
+#define ADDR                (dev->params.addr)
+
+int sht2x_init(sht2x_t *dev, const sht2x_params_t *params)
+{
+    uint8_t result;
+    uint8_t user_reg;
+
+    assert(dev && params);
+    assert(params->resolution == SHT2x_RES_12_14BIT || 
+           params->resolution == SHT2x_RES_8_12BIT || 
+           params->resolution == SHT2x_RES_10_13BIT || 
+           params->resolution == SHT2x_RES_11_11BIT );
+
+    /* write device descriptor */
+    DEBUG("[sht2x] init: copying dev descriptor\n");
+    memcpy(dev, params, sizeof(sht2x_params_t));
+
+    /* initialize the I2C bus */
+    DEBUG("[sht2x] init: acquiring I2C\n");
+    i2c_acquire(BUS);
+    DEBUG("[sht2x] init: setting master mode\n");
+    if (i2c_init_master(BUS, I2C_SPEED) < 0) {
+        i2c_release(BUS);
+        DEBUG("[sht2x] init - error: unable to initialize I2C bus\n");
+        return SHT2X_NOI2C;
+    }
+
+    // reset the device - send a soft reset command
+    DEBUG("[sht2x] init: issuing soft reset\n");
+    result = i2c_write_byte(BUS, ADDR, SHT2X_SOFT_RESET);
+    if (result < 1) {
+        i2c_release(BUS);
+        DEBUG("[sht2x] init - error: soft reset failed\n");
+        return SHT2X_NODEV;
+    }
+
+    // wait 15 ms for device to reset
+    DEBUG("[sht2x] init: waiting\n");
+    i2c_release(BUS);
+    xtimer_usleep(15 * 1000);
```

Comment:
Even better:

- make `15` a configurable macro/define
- use `US_PER_MS` instead of `1000`